demos/sample_azure_iot_adu/sample_azure_iot_pnp_simulated_data.c

```diff
@@ -277,9 +279,9 @@ static bool prvIsUpdateAlreadyInstalled( const AzureIoTADUUpdateRequest_t * pxAd
  */
 static AzureIoTADURequestDecision_t prvUserDecideShouldStartUpdate( AzureIoTADUUpdateRequest_t * pxAduUpdateRequest )
 {
-    if( prvIsUpdateAlreadyInstalled( pxAduUpdateRequest ) )
+    if( !prvDoesInstalledCriteriaMatchCurrentVersion( pxAduUpdateRequest ) )
```

Comment:
If it does match, the update manifest was applied and the device rebooted correct?
```suggestion
    if( prvDoesInstalledCriteriaMatchCurrentVersion( pxAduUpdateRequest ) )
```

Reply:
If the device rebooted, now it has the new version (1.1), so the "installedCriteria" will not match anymore. The name from ADU may be misleading, I believe.

Reply:
https://docs.microsoft.com/en-us/previous-versions/azure/iot-hub-device-update/import-schema#import-manifest-schema

From there ^

> String interpreted by the agent to determine whether the update was applied successfully

I'm assuming that is a string that is of importance for the device to compare in its own way. If that "installedCriteria" matches, that means it updated successfully.

Reply:
I think if it's the way you're thinking, it would be called `installCriteria`? In that case if it matched, it would mean apply the update.